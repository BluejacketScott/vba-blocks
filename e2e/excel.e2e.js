const { resolve, join } = require('path');
const { tmp, execute, check } = require('./helpers/execute');
const { openExcel, closeExcel } = require('./helpers/addin');
const { copyFile } = require('../lib/utils/fs');
const { default: createTarget } = require('../lib/targets/create-target');

jest.setTimeout(10000);

beforeAll(openExcel);
afterAll(closeExcel);

test('build', async () => {
  const dir = resolve(__dirname, `./fixtures/standard`);
  const { path: cwd, cleanup } = await tmp(dir);

  try {
    await execute(cwd, 'build');

    const result = await check(cwd);
    expect(result).toMatchSnapshot();
  } finally {
    await cleanup();
  }
});

test('import', async () => {
  const dir = resolve(__dirname, `./fixtures/standard`);
  const { path: cwd, cleanup } = await tmp(dir);

  try {
    const project = {
      paths: { build: join(cwd, 'build') }
    };
    const target = {
      name: 'e2e-standard',
      path: join(cwd, 'targets/xlsm'),
      filename: 'e2e-standard.xlsm'
    };

    await createTarget(project, target);
    await execute(cwd, 'import');

    const result = await check(cwd);
    expect(result).toMatchSnapshot();
  } finally {
    await cleanup();
  }
});

test('export', async () => {
  const empty = resolve(__dirname, './fixtures/empty');
  const std = resolve(__dirname, './fixtures/standard');

  const [
    { path: cwdEmpty, cleanup: cleanupEmpty },
    { path: cwdStd, cleanup: cleanupStd }
  ] = await Promise.all([tmp(empty), tmp(std)]);

  try {
    // First, build projects
    await execute(cwdEmpty, 'build');
    await execute(cwdStd, 'build');

    // Move standard to empty (new things to export)
    // TODO only checks add, need to check change and remove
    await copyFile(
      join(cwdStd, 'build/e2e-standard.xlsm'),
      join(cwdEmpty, 'build/e2e-empty.xlsm')
    );

    await execute(cwdEmpty, 'export xlsm');

    const result = await check(cwdEmpty);
    expect(result).toMatchSnapshot();
  } finally {
    await Promise.all([cleanupEmpty(), cleanupStd()]);
  }
});
