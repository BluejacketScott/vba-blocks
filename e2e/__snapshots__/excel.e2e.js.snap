// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build 1`] = `
Object {
  "build/e2e-standard.xlsm": "<TODO>",
  "src/A.bas": "7530e3fbdb802143df3c9302dbf6a74b4542b30cd0fd0634796b8a65fddba45f",
  "src/B.cls": "04f1372b93222974d1a64dc571241589fa3e09ec628506d647a22562bfa2d7f6",
  "targets/xlsm/[Content_Types].xml": "1201ce9d35181763181e983a4fc7a947695e8c8659eff83e0d47d39aef2f2d49",
  "targets/xlsm/_rels/.rels": "73e5a29f48d5ab979eeda062493bc7e679265c1344ef936978b8becec5549497",
  "targets/xlsm/docProps/app.xml": "487796376309f7eed5ba21ce8a9f55395787862d8121c85cdb7258b128baa0f8",
  "targets/xlsm/docProps/core.xml": "1d4e563a25803684e1973d771cc8393b0fb15ed7bbbd547a97daf6e5f3924e58",
  "targets/xlsm/xl/_rels/workbook.xml.rels": "353383fbe90f202be596bdf0a86439c704d7024f4f8c19ebc9d06b42a341dba1",
  "targets/xlsm/xl/sharedStrings.xml": "3dbe932e2d7a4c57b784cccc469ebd14b0a80e6efc8fa71bf2dfb498c746ed5e",
  "targets/xlsm/xl/styles.xml": "f2608cf432cf5939abe03f4fc8545f59c8632dd50d2255ae89525652ab6cda07",
  "targets/xlsm/xl/theme/theme1.xml": "b16106f4229f6ece6d74426b3e73ff6a677df2908d2a06f76ef0329fa27c0282",
  "targets/xlsm/xl/workbook.xml": "8e72e019326a2b3348c6865e66c248c751f99945dea6bdb53bca29366a3e0fa6",
  "targets/xlsm/xl/worksheets/sheet1.xml": "4e256f9a068483eec8facf4a640938e507905642b5248a22d487b491786fa93f",
  "vba-block.lock": "f0055813646aff3962104a0e9ceba138ffaf12f49774bc11d3136541bbe66458",
  "vba-block.toml": "252a6666582be23dfde1e088d5b869f947377f84d961280a3e3ea2a6304dd7e4",
}
`;

exports[`export 1`] = `
Object {
  "A.bas": "7530e3fbdb802143df3c9302dbf6a74b4542b30cd0fd0634796b8a65fddba45f",
  "B.cls": "04f1372b93222974d1a64dc571241589fa3e09ec628506d647a22562bfa2d7f6",
  "Sheet1.cls": "53279ec0e31ca2ec31267d871654588dcbda97343d3e2e6b2487fedda858c0ba",
  "ThisWorkbook.cls": "ffc5d3e2a71c29a2c69fc82c77d3d11b2dbc3cc6111e3e0074b6af7e7e689639",
  "build/e2e-empty.xlsm": "<TODO>",
  "targets/xlsm/[Content_Types].xml": "1201ce9d35181763181e983a4fc7a947695e8c8659eff83e0d47d39aef2f2d49",
  "targets/xlsm/_rels/.rels": "73e5a29f48d5ab979eeda062493bc7e679265c1344ef936978b8becec5549497",
  "targets/xlsm/docProps/app.xml": "487796376309f7eed5ba21ce8a9f55395787862d8121c85cdb7258b128baa0f8",
  "targets/xlsm/docProps/core.xml": "1d4e563a25803684e1973d771cc8393b0fb15ed7bbbd547a97daf6e5f3924e58",
  "targets/xlsm/xl/_rels/workbook.xml.rels": "353383fbe90f202be596bdf0a86439c704d7024f4f8c19ebc9d06b42a341dba1",
  "targets/xlsm/xl/sharedStrings.xml": "3dbe932e2d7a4c57b784cccc469ebd14b0a80e6efc8fa71bf2dfb498c746ed5e",
  "targets/xlsm/xl/styles.xml": "f2608cf432cf5939abe03f4fc8545f59c8632dd50d2255ae89525652ab6cda07",
  "targets/xlsm/xl/theme/theme1.xml": "b16106f4229f6ece6d74426b3e73ff6a677df2908d2a06f76ef0329fa27c0282",
  "targets/xlsm/xl/workbook.xml": "8e72e019326a2b3348c6865e66c248c751f99945dea6bdb53bca29366a3e0fa6",
  "targets/xlsm/xl/worksheets/sheet1.xml": "4e256f9a068483eec8facf4a640938e507905642b5248a22d487b491786fa93f",
  "vba-block.lock": "77dd85aed5bfeb1bc6195fd5ce4e5c5eb13bdfcab746ea072b4b3fc090c89849",
  "vba-block.toml": "260dd08a1e514115c543e29b3b8948d5578a24d2e4274fc7093d4dd8e12ad788",
}
`;

exports[`import 1`] = `
Object {
  "build/e2e-standard.xlsm": "<TODO>",
  "src/A.bas": "7530e3fbdb802143df3c9302dbf6a74b4542b30cd0fd0634796b8a65fddba45f",
  "src/B.cls": "04f1372b93222974d1a64dc571241589fa3e09ec628506d647a22562bfa2d7f6",
  "targets/xlsm/[Content_Types].xml": "1201ce9d35181763181e983a4fc7a947695e8c8659eff83e0d47d39aef2f2d49",
  "targets/xlsm/_rels/.rels": "73e5a29f48d5ab979eeda062493bc7e679265c1344ef936978b8becec5549497",
  "targets/xlsm/docProps/app.xml": "487796376309f7eed5ba21ce8a9f55395787862d8121c85cdb7258b128baa0f8",
  "targets/xlsm/docProps/core.xml": "1d4e563a25803684e1973d771cc8393b0fb15ed7bbbd547a97daf6e5f3924e58",
  "targets/xlsm/xl/_rels/workbook.xml.rels": "353383fbe90f202be596bdf0a86439c704d7024f4f8c19ebc9d06b42a341dba1",
  "targets/xlsm/xl/sharedStrings.xml": "3dbe932e2d7a4c57b784cccc469ebd14b0a80e6efc8fa71bf2dfb498c746ed5e",
  "targets/xlsm/xl/styles.xml": "f2608cf432cf5939abe03f4fc8545f59c8632dd50d2255ae89525652ab6cda07",
  "targets/xlsm/xl/theme/theme1.xml": "b16106f4229f6ece6d74426b3e73ff6a677df2908d2a06f76ef0329fa27c0282",
  "targets/xlsm/xl/workbook.xml": "8e72e019326a2b3348c6865e66c248c751f99945dea6bdb53bca29366a3e0fa6",
  "targets/xlsm/xl/worksheets/sheet1.xml": "4e256f9a068483eec8facf4a640938e507905642b5248a22d487b491786fa93f",
  "vba-block.lock": "f0055813646aff3962104a0e9ceba138ffaf12f49774bc11d3136541bbe66458",
  "vba-block.toml": "252a6666582be23dfde1e088d5b869f947377f84d961280a3e3ea2a6304dd7e4",
}
`;
